server.go

```diff
@@ -1316,20 +1331,28 @@ func (s *server) handleRelayInvMsg(state *peerState, msg relayMsg) {
 			if sp.relayTxDisabled() {
 				return
 			}
+
+			txD, ok := msg.data.(*mempool.TxDesc)
+			if !ok {
+				peerLog.Warnf("Underlying data for tx" +
+					" inv relay is not a *mempool.TxDesc")
+				return
+			}
+
 			// Don't relay the transaction if there is a bloom
 			// filter loaded and the transaction doesn't match it.
 			if sp.filter.IsLoaded() {
-				txD, ok := msg.data.(*mempool.TxDesc)
-				if !ok {
-					peerLog.Warnf("Underlying data for tx" +
-						" inv relay is not a *mempool.TxDesc")
-					return
-				}
-
 				if !sp.filter.MatchTxAndUpdate(txD.Tx) {
 					return
 				}
 			}
+
+			// Don't relay the transaction if the transaction fee-per-kb
+			// is less than the peer's feefilter.
+			feeFilter := atomic.LoadInt64(&sp.feeFilter)
```

Comment:
Should this come before the bloom filter check?  I mention it because it's cheaper, and, as a general rule, it's typically better to save the more expensive checks to last so they can be avoided when the cheaper checks don't pass.